src/MultiFormats/Cryptography/KeccakManaged.cs

```diff
@@ -0,0 +1,332 @@
+﻿// The SHA3 doesn't create .Net Standard package.
+// This is a copy of https://bitbucket.org/jdluzen/sha3/raw/d1fd55dc225d18a7fb61515b62d3c8f164d2e788/SHA3Managed/SHA3Managed.cs
+
+using System;
+
+namespace MultiFormats.Cryptography
+{
+    internal partial class KeccakManaged : Keccak
+    {
+        public KeccakManaged(int hashBitLength)
+            : base(hashBitLength) { }
+
+        protected override void HashCore(byte[] array, int ibStart, int cbSize)
+        {
+            base.HashCore(array, ibStart, cbSize);
+            if (cbSize == 0)
+                return;
+            var sizeInBytes = SizeInBytes;
+            if (buffer == null)
+                buffer = new byte[sizeInBytes];
+            var stride = sizeInBytes >> 3;
+            var utemps = new ulong[stride];
+            if (buffLength == sizeInBytes)
+                throw new Exception("Unexpected error, the internal buffer is full");
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: ['System.Exception' should not be thrown by user code.](https://app.codacy.com/app/catalyst-network/Catalyst.Framework/pullRequest?prid=4612924)